courses-rest/src/main/java/no/sikt/nva/fs/CoursesProvider.java

```diff
@@ -0,0 +1,110 @@
+package no.sikt.nva.fs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.text.Collator;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import no.sikt.nva.fs.client.FsClient;
+import no.sikt.nva.fs.client.FsClientException;
+import no.sikt.nva.fs.client.FsCourse;
+import no.sikt.nva.fs.client.FsSemester;
+import no.sikt.nva.fs.client.HttpUrlConnectionFsClient;
+import no.sikt.nva.fs.client.Item;
+import no.sikt.nva.fs.config.InstitutionConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CoursesProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CoursesProvider.class);
+    /* default */ static final String LOG_MESSAGE_PREFIX_FS_COMMUNICATION_PROBLEM =
+        "Unable to communicate with FS API for ";
+
+    private final ObjectMapper objectMapper;
+    private final String fsBaseUri;
+    private final InstitutionConfig institutionConfig;
+
+    public CoursesProvider(final ObjectMapper objectMapper,
+                           final String fsBaseUri,
+                           final InstitutionConfig institutionConfig) {
+
+        this.objectMapper = objectMapper;
+        this.fsBaseUri = fsBaseUri;
+        this.institutionConfig = institutionConfig;
+    }
+
+    public Course[] getCurrentlyTaughtCourses(final int year, final int month) {
+
+        LOGGER.debug("Fetching courses by institution '{}' from FS", institutionConfig.getCode());
+
+        final FsClient fsClient = new HttpUrlConnectionFsClient(objectMapper,
+                                                                fsBaseUri,
+                                                                institutionConfig.getCode(),
+                                                                institutionConfig.getUsername(),
+                                                                institutionConfig.getPassword());
+
+        final Map<Integer, List<Term>> yearsToTerms = new ConcurrentHashMap<>();
+        /*
+            During the first half of the year, we look up all courses for the current year. In the last half of the
+            year, we look up courses for the rest of the current year and the first half of the next year:
```

Comment:
This important piece of business logic should be more apparent in the tests, i.e. it should be included in the test names.
Right now, one can only extrapolate it based on the values that are expected by the tests

Reply:
Improved name of test methods to better reflect expectations